ui/components/app/assets/nfts/nft-options/nft-options.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { useContext, useRef, useState } from 'react';
+import { I18nContext } from '../../../../../contexts/i18n';
+import {
+  Box,
+  ButtonIcon,
+  ButtonIconSize,
+  Icon,
+  IconName,
+  IconSize,
+  Popover,
+  PopoverPosition,
+} from '../../../../component-library';
+import { IconColor } from '../../../../../helpers/constants/design-system';
+import { SelectableListItem } from '../../asset-list/sort-control/sort-control';
+
+type NftOptionsProps = {
+  onRemove: () => void;
+  onViewOnOpensea?: () => void;
+  showOpenSeaLink: boolean;
+};
+
+const NftOptions = ({
+  onRemove,
+  onViewOnOpensea,
+  showOpenSeaLink,
+}: NftOptionsProps) => {
+  const t = useContext(I18nContext);
+  const [nftOptionsOpen, setNftOptionsOpen] = useState(false);
+  const ref = useRef<HTMLElement | null>(null);
+
+  const closePopover = () => {
+    setNftOptionsOpen(false);
+    onViewOnOpensea && onViewOnOpensea();
+  };
+
+  return (
+    <Box ref={ref}>
+      <ButtonIcon
+        iconName={IconName.MoreVertical}
+        data-testid="nft-options__button"
+        onClick={() => setNftOptionsOpen(true)}
+        color={IconColor.iconDefault}
+        size={ButtonIconSize.Sm}
+        ariaLabel={t('nftOptions')}
+      />
+      <Popover
+        onClickOutside={closePopover}
+        isOpen={nftOptionsOpen}
+        position={PopoverPosition.BottomEnd}
+        referenceElement={ref.current}
+        matchWidth={false}
+        style={{
+          zIndex: 10,
+          display: 'flex',
+          flexDirection: 'column',
+          padding: 0,
+        }}
+      >
+        {showOpenSeaLink ? (
+          <SelectableListItem
+            testId="nft-options__view-on-opensea"
+            onClick={() => {
+              setNftOptionsOpen(false);
+              onViewOnOpensea && onViewOnOpensea();
```

Comment:
nit - This is the same as `closePopover`, is that intentional?

Should the `closePopover` above when clicking outside do the navigation to opensea?

Reply:
Nice catch, this is a bug. Will fix 👍

Reply:
https://github.com/MetaMask/metamask-extension/pull/29445/commits/dd50a1b5d1c44f8bf6f33d26a18a1e14ee328d71